### PR TITLE
Implement binary expressions in the new query engine

### PR DESCRIPTION
**Pull request by jsternberg**
Binary expressions for float iterators have been implemented. Binary
expressions for other operations, like string, integer, and boolean,
still need to be implemented.

### CODE REVIEW

File: influxql/select.go [hunk 1]
```diff
@@ -199,11 +204,266 @@ func buildExprIterator(expr Expr, ic IteratorCreator, opt IteratorOptions) (Iter
...
+	case FloatIterator:
+		return Float
+	case StringIterator:
+		return String
+	case BooleanIterator:
+		return Boolean
+	default:
+		return Unknown
+	}
+}
```
Comment (benbjohnson) [inline influxql/select.go:430]: Can you change the function name from `getIteratorDataType()` to `iteratorDataType()`?

https://golang.org/doc/effective_go.html#Getters

File: influxql/select.go [hunk 2]
```diff
@@ -199,11 +204,266 @@ func buildExprIterator(expr Expr, ic IteratorCreator, opt IteratorOptions) (Iter
...
+				return lhs > rhs
+			}
+		case GTE:
+			return func(lhs, rhs float64) bool {
+				return lhs >= rhs
+			}
+		}
+	}
+	return nil
+}
```
Comment (benbjohnson) [inline influxql/select.go:482]: Change the name to `binaryExprFunc()`.

https://golang.org/doc/effective_go.html#Getters

**Comment by benbjohnson** [comment]
Overall lgtm

File: influxql/select.go (hunk 1 above)
Comment (jsternberg) [inline influxql/select.go:430]: Done.

File: influxql/select.go (hunk 2 above)
Comment (jsternberg) [inline influxql/select.go:482]: Done.

**Pull request merged**